src/test/java/org/eclipse/uprotocol/communication/InMemorySubscriberTest.java

```diff
@@ -46,21 +49,37 @@ public void onReceive(UMessage message) {
         assertFalse(response.toCompletableFuture().isCompletedExceptionally());
     }
 
+    @Test
+    @DisplayName("Test subscribe and then unsubscribe happy path")
+    public void test_subscribe_and_then_unsubscribe_happy_path() {
+        UUri topic = createTopic();
+        UTransport transport = new TestUTransport();
+        UListener listener = new UListener() {
+            @Override
+            public void onReceive(UMessage message) {
+                // Do nothing
+            }
+        };
+        Subscriber subscriber = new InMemorySubscriber(transport, new InMemoryRpcClient(transport));
```

Comment:
Probably not a good idea to test both ImMemoryRpcClient and InMemorySubscriber in the same unit test. Suggest to mock InMemoryRpcClient with mocking library like mockito

Reply:
Good point. I will take this for the next PR. Can you point me to an example so I can see?